### ISSUE TITLE
Support for Keepass 'Duplicate Entry' mechanism with references to original credentials

### ISSUE THREAD

**Issue by flaklypa** [perfectapi/CKP#28]
Keepass supports creating a duplicate of a certain entry AND populate the username/password with references to the original, duplicated entry. This allows for using the same username/password combo for multiple sites (URLs) while only specifying the username/password once. 
It would be nice if CKP could deal with these references.

**Comment by DukeyToo** [comment]
I was not aware of that - I'll look into it

**Comment by shawnholt** [comment]
Would this better support multiple accounts on the same URL (like google?)  I use lastpass and the url /matching is not very good.

**Comment by DukeyToo** [comment]
+1 from support tab on web store: "Would be great if it could utilize field references. It outputs the references themselves instead of the referenced data. I use them heavily."

Application version: 2015.5.4
Client version: 5.0 (X11; CrOS armv7l 6812.83.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/42.0.2311.134 Safari/537.36
Reported date: 5/12/15 8:34 PM
Reported by: A Google User

**Comment by DukeyToo** [comment]
@shawnholt I don't use this feature of KeePass myself, but I think it is more for scenarios where you use the same username or password on multiple sites.  

CKP is already pretty good at matching to the correct website, and supports the scenario you describe (multiple credentials for the same site).

**Comment by NeoSelcev** [comment]
+1
@DukeyToo I'm using PC KeePass all the time, and CKP for a while.

Lets say, you have a google account (or an apple id). For google, you might need to log in on several web sites, like google itself, youtube, gmail, calendar, bookmarks, passwords... and many others with the same credentials. For Apple, you might need the same credentials for itunes store, iCloud, findmyiphone, and more.

KeePass naturally doesn't support multiple URLs for the single entity, but it does support referencing the credentials. Due the organization purposes you might want to have single account source for such web services, like google, and apple, and make references to them, when creating logins entities for each site. In this case you can change username or password only once inside the main account entity in KeePass instead of changing them for all, lets say, google services...

So It would be really great to have such feature working in CKP too. It would be very very VERY usefull.

**Comment by DukeyToo** [comment]
http://keepass.info/help/base/fieldrefs.html

**Comment by DukeyToo** [comment]
Just an update on this - I have a working version for basic references in a different branch, but have not merged yet because there are many reference scenarios not yet supported, and I haven't had time to finish the feature.  If there is support for releasing the unfinished version, let me know in this thread and I'll do so.

**Comment by MikeDR667** [comment]
yes please I use references allot

**Comment by NeoSelcev** [comment]
+1

**Comment by trashcanpatrol** [comment]
+1, could we please have an update on this? I'd love to be able to use field references.

**Comment by laiskimus** [comment]
+1

**Comment by robbratton** [comment]
+1 on partial support for references. I'd be willing to beta test.

**Comment by cmccandless** [comment]
Can we get an update on this? This is an essential feature for users of SSO systems (such as college students) where the same username and password are used across multiple domains.

**Comment by shikhanshu** [comment]
Any update on this? As of 4/13/2017, the latest CKP still outputs reference string instead of the referenced value.
Alternatively, how can I make CKP always use a specific entry for a particular URL? For example, in my KeePass DB, I have "mint.intuit.com" and a duplicate key "turbotax.intuit.com". If I can tell CKP to pick up the "mint.intuit.com" credentials when visiting "turbotax.intuit.com", I can just go ahead and delete "turbotax.intuit.com" from my KeePass DB.

**Comment by pmajkutewicz** [comment]
+1

**Comment by jeffbowman** [comment]
Interestingly, on my Mac, if I choose to 'find manually' instead of using the matched entry, it seems the referenced values are output for both User and Password, but if I just use the matched one based on the url, the password is the reference text (ie {REF:P@I:ABCDEF123456789ABCDEF123456790AB}). My version is 2017.5.30. So, it seems to partially work.

**Comment by subdavis** [comment]
There's actually a beautiful bug here that comes into play when UUID's Hex byte representations get cast to strings.

For example, if your byte was 00000001, that should be 0x01 or "01", but it's cast to "1".

I'll submit a patch real soon - it's a one liner :)
